packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin2.test.tsx

```diff
@@ -0,0 +1,221 @@
+import uPlot from 'uplot';
+
+import { getFeatureToggle } from '../../../utils/featureToggle';
+
+import { setupXAxisPan } from './TooltipPlugin2';
+
+jest.mock('../../../utils/featureToggle', () => ({
+  getFeatureToggle: jest.fn(),
+}));
+
+const mockGetFeatureToggle = getFeatureToggle as jest.MockedFunction<typeof getFeatureToggle>;
+
+const createMockXAxis = () => {
+  const element = document.createElement('div');
+  element.classList.add('u-axis');
+  return element;
+};
+
+const createMockUPlot = (xAxisElement: HTMLElement): Partial<uPlot> => {
+  const root = document.createElement('div');
+  root.appendChild(xAxisElement);
+
+  const over = document.createElement('div');
+  Object.defineProperty(over, 'getBoundingClientRect', {
+    value: () => ({ left: 0, top: 0, width: 800, height: 400 }),
+  });
+
+  return {
+    root,
+    over,
+    bbox: { width: 800, height: 400, left: 0, top: 0 },
+    scales: {
+      x: {
+        min: 1000,
+        max: 2000,
+        range: () => [1000, 2000],
+      },
+    },
+    setScale: jest.fn(),
+  } as Partial<uPlot>;
+};
+
+describe('TooltipPlugin2', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('setupXAxisPan', () => {
+    let mockQueryZoom: jest.Mock;
+    let xAxisElement: HTMLElement;
+    let mockUPlot: Partial<uPlot>;
```

Comment:
Lot of casts to uPlot throughout the mocks that I think we can avoid by doing the cast up top.
```suggestion
    let mockUPlot: uPlot;
```